pkg/internal/checkup/trex/client.go

```diff
@@ -32,3 +46,111 @@ func NewClient(vmiSerialClient vmiSerialConsoleClient, namespace string, verbose
 		verbosePrintsEnabled: verbosePrintsEnabled,
 	}
 }
+
+func (t Client) WaitTrexServerReady(ctx context.Context, vmiName string) error {
+	const (
+		interval = 5 * time.Second
+		timeout  = time.Minute
+	)
+	var err error
+	ctxWithNewDeadline, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	conditionFn := func(ctx context.Context) (bool, error) {
+		if t.isTrexServerRunning(vmiName) {
+			log.Printf("trex-server is now ready")
+			return true, nil
+		}
+		if t.verbosePrintsEnabled {
+			log.Printf("trex-server is not ready yet...")
+		}
+		return false, nil
+	}
+	if err = wait.PollImmediateUntilWithContext(ctxWithNewDeadline, interval, conditionFn); err != nil {
+		if errors.Is(err, wait.ErrWaitTimeout) {
+			if t.verbosePrintsEnabled {
+				if logErr := t.printTrexServiceFailLogs(vmiName); logErr != nil {
+					return logErr
+				}
+			}
+			return fmt.Errorf("timeout waiting for trex-server to be ready")
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func (t Client) isTrexServerRunning(vmiName string) bool {
+	const helpSubstring = "Console Commands"
+	resp, err := t.runTrexConsoleCmd(vmiName, "help")
+	if err != nil || !strings.Contains(resp, helpSubstring) {
+		return false
+	}
+	return true
+}
+
+func (t Client) printTrexServiceFailLogs(vmiName string) error {
+	var err error
+	trexServiceStatus, err := t.getTrexServiceStatus(vmiName)
+	if err != nil {
+		return fmt.Errorf("failed gathering systemctl service status after trex-server timeout: %w", err)
+	}
+	trexJournalctlLogs, err := t.getTrexServiceJournalctl(vmiName)
+	if err != nil {
+		return fmt.Errorf("failed gathering trex.service related joutnalctl logs after trex-server timeout: %w", err)
+	}
+	log.Printf("timeout waiting for trex-server to be ready\n"+
+		"systemd service status:\n%s\n"+
+		"joutnalctl logs:\n%s", trexServiceStatus, trexJournalctlLogs)
+	return nil
+}
+
+func (t Client) getTrexServiceStatus(vmiName string) (string, error) {
+	command := fmt.Sprintf("systemctl status %s | cat", SystemdUnitFileName)
+	resp, err := console.SafeExpectBatchWithResponse(t.vmiSerialClient, t.namespace, vmiName,
+		[]expect.Batcher{
+			&expect.BSnd{S: command + "\n"},
+			&expect.BExp{R: shellPrompt},
+		},
+		batchTimeout,
+	)
+	return resp[0].Output, err
+}
+
+func (t Client) getTrexServiceJournalctl(vmiName string) (string, error) {
+	command := fmt.Sprintf("journalctl | grep %s", SystemdUnitFileName)
+	resp, err := console.SafeExpectBatchWithResponse(t.vmiSerialClient, t.namespace, vmiName,
+		[]expect.Batcher{
+			&expect.BSnd{S: command + "\n"},
+			&expect.BExp{R: shellPrompt},
+		},
+		batchTimeout,
+	)
+	return resp[0].Output, err
+}
+
+func (t Client) runTrexConsoleCmd(vmiName, command string) (string, error) {
+	shellCommand := fmt.Sprintf("/bin/sh -c \"cd /opt/trex && echo %q | ./trex-console -q\"", command)
```

Comment:
Why do we need `/bin/sh -c \"cd /opt/trex &&` for every command?
We could `cd` into the trex dir from executor once.